native-image-support/src/main/resources/META-INF/native-image/com.google.cloud/google-cloud-graalvm-support/proxy-config.json

```diff
@@ -0,0 +1,14 @@
+[
```

Comment:
I think a better home for this is in cloud-sql-sample/resources/META-INF/native-image/com.example. The current location works fine when we call `mvn package -P native` from within the `cloud-sql-sample` directory but it is currently breaking unrelated samples with the following error: `UserError$UserException: Class jnr.unixsocket.Native$LibC`. Here is a example of how we define proxy-config.json in [another sample.](https://github.com/GoogleCloudPlatform/native-image-support-java/tree/master/native-image-samples/native-image-samples-spring/spring-cloud-gcp-trace-sample/src/main/resources/META-INF/native-image/com.example)

Reply:
I'm confused, as a library, we have to include proxy-config.json in the final output artifact (native-image-support.jar) in order for users of this library to be able to use unix domain sockets.

If only have proxy-config.json in the samples, it means that end users need to copy proxy-config.json to their projects like this sample to use.

Reply:
I found the reason, unlike resource file, which can be specified by pattern, so the file does not exist and can be skipped automatically. The proxy must be specified to the certain class name, if there is no such class, it will cause native-image error.

I'm trying to find a way to fix this.

Reply:
i delete proxy-config.json and register dynamic proxy in code, thus will not influence other case which don't have `mysql-socket-factory-connector` in classpath